### PR TITLE
[Feature] Add TD-MPC2 Q-network ensemble

### PR DESCRIPTION
**Pull request by bsprenger**
## Description

> [!NOTE]
> This PR stacks on #4452 . I cannot find a good way to compare the diff against that base, since it is in my own fork.
> Once it is merged, I will re-target this one to make the diff easier to read.

This PR introduces the TD-MPC2 Q ensemble, which is a key part of the algorithm. The Q-ensemble consists of $N$ parallel Q-functions to compute TD targets during training of the model and to be used during the planning stage of the algorithm.

## Motivation and Context

This is the 4th PR in the chain intended to address #4448 . The Q-ensemble is described in depth in the [original paper](https://arxiv.org/pdf/2310.16828) and associated [implementation](https://github.com/nicklashansen/tdmpc2).

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

- [x] New feature (non-breaking change which adds core functionality)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/trainers/algorithms/tdmpc2.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    :meth:`forward` returns the logits of every Q-function. :meth:`reduce`
+    follows TD-MPC2's value-estimation rule by selecting two Q-functions at
+    random, decoding their logits, and taking either their minimum or average.
+    Online, detached, and target parameter sources are available for both
+    operations.
+
+    Args:
+        q_networks: Sequence of identically shaped single-network Q-functions.
+        num_bins: Number of categorical bins. ``0`` uses an identity output and
+            ``1`` uses a symmetric exponential output.
```
Comment (vmoens) [inline torchrl/trainers/algorithms/tdmpc2.py]: Advertised num_bins modes are rejected.

The Q-ensemble documentation says 0 provides identity regression and 1 provides symlog regression, but the [constructor rejects both](https://github.com/pytorch/rl/pull/4453/files). The config factories and loss also only support categorical mode.

Either implement the documented/reference behavior or explicitly narrow the API and documentation to num_bins > 1.

Comment (bsprenger) [reply torchrl/trainers/algorithms/tdmpc2.py]: I updated the documented behaviour. Thanks!
